motionplan/motionPlanner.go

```diff
@@ -54,10 +54,49 @@ type PlanRequest struct {
 	Options            map[string]interface{}
 }
 
+// PlanRequestValidation ensures PlanRequests are not malformed.
+func PlanRequestValidation(req PlanRequest) (*PlanRequest, error) {
+	if req.Logger == nil {
+		req.Logger = golog.NewLogger("plan-request-logger")
+		req.Logger.Info("no logger was given, so we created one")
+	}
+	if req.Goal == nil {
+		return &req, errors.New("PlanRequest cannot have nil goal")
+	}
+	if req.Frame == nil {
+		return &req, errors.New("PlanRequest cannot have nil frame")
+	}
+	if req.FrameSystem == nil {
+		req.Logger.Info("no FrameSystem was given, so we ephemerally create a framesystem containing just the frame for the solve")
+		fs := frame.NewEmptyFrameSystem("")
+		if err := fs.AddFrame(req.Frame, fs.World()); err != nil {
+			return &req, err
+		}
+		req.FrameSystem = fs
+	} else if req.FrameSystem.Frame(req.Frame.Name()) == nil {
+		return &req, errors.New("FrameSystem of PlanRequest does not contain the Frame of PlanRequest")
+	}
+	if len(req.StartConfiguration) == 0 {
+		return &req, errors.New("PlanRequest cannot have nil StartConfiguration")
+	}
+	if req.WorldState == nil {
+		req.Logger.Info("no WorldState was given, so we created one")
```

Comment:
This should also be a `Debug`

Reply:
I don't think we should log this

Reply:
I think this should remain for the same reason as above, curious to hear what others think.